metaflow/plugins/argo/argo_workflows_deployer.py

```diff
@@ -188,6 +225,59 @@ def delete(instance: DeployedFlow, **kwargs):
     return command_obj.process.returncode == 0
 
 
+def from_deployment(identifier: str, metadata: str = None):
+    client = ArgoClient(namespace=KUBERNETES_NAMESPACE)
+    workflow_template = client.get_workflow_template(identifier)
+
+    metadata_annotations = workflow_template.get("metadata", {}).get("annotations", {})
+
+    flow_name = metadata_annotations.get("metaflow/flow_name", "")
+    parameters = json.loads(metadata_annotations.get("metaflow/parameters", {}))
+
+    # these two only exist if @project decorator is used..
+    branch_name = metadata_annotations.get("metaflow/branch_name", None)
+    project_name = metadata_annotations.get("metaflow/project_name", None)
+
+    project_kwargs = {}
+    if branch_name is not None:
+        if branch_name.startswith("prod."):
+            project_kwargs["production"] = True
+            project_kwargs["branch"] = branch_name[len("prod.") :]
+        elif branch_name.startswith("test."):
+            project_kwargs["branch"] = branch_name[len("test.") :]
+        elif branch_name == "prod":
```

Comment:
what happens if i submit a flow with `--branch=prod`?

Reply:
branch name will be `test.prod` in that case..
and we already handle that case.. so we are good..

in case `--branch=test`, branch name will be `test.test` in that case.. this is also covered..

in case of `--branch=prod` and `--production`, branch name will be `prod.prod` and this case is also covered by the above logic..